### ISSUE TITLE
[24.4 beta4] having more than one `@index` breaks the application

### ISSUE THREAD

**Issue by taefi** [vaadin/hilla#2460]
### Describe the bug

Having more than one `@index` breaks the application loading on the browser due to having duplicate keys in the `RouteUnifyingIndexHtmlRequestListener`:
```
java.lang.IllegalStateException: Duplicate key  (attempted merging values AvailableViewInfo{title='Index', rolesAllowed=null, requiresLogin=false, route='', lazy=false, register=false, menu=null, routeParameters={}} and AvailableViewInfo{title='Number Signal', rolesAllowed=null, requiresLogin=false, route='', lazy=false, register=false, menu=ClientViewMenuConfig[title=Number Signal, order=0.0, icon=line-awesome/svg/globe-solid.svg, exclude=null], routeParameters={}})
	at java.base/java.util.stream.Collectors.duplicateKeyException(Collectors.java:135) ~[na:na]
	at java.base/java.util.stream.Collectors.lambda$uniqKeysMapAccumulator$1(Collectors.java:182) ~[na:na]
	at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179) ~[na:na]
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133) ~[na:na]
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1939) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[na:na]
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[na:na]
	at com.vaadin.hilla.route.RouteUnifyingIndexHtmlRequestListener.collectClientViews(RouteUnifyingIndexHtmlRequestListener.java:154) ~[hilla-endpoint-24.4.0.beta4.jar:na]
```

### Expected-behavior

It should be possible to have `@index.tsx` files in as many desired directories as possible under `frontend/views/`.

### Reproduction

A directory structure like this would reveal this bug:
```
views/
|_ numbers/
|   |_ @index.tsx
|_ @index.tsx
```
The reason is that the Map keys are created based on the `route` property of ClientViewConfig objects, instead of their actual route in the hierarchy:
![Screenshot 2024-05-23 at 15 26 55](https://github.com/vaadin/hilla/assets/70698630/324dc578-12d1-48f3-bdba-6ac6427535cd)

### System Info

Vaadin: 24.4.0.beta4

**Comment by taefi** [comment]
The above mentioned issue with the way client views menu data is collected in a map with wrong keys, makes the menu navigation targets to be defined wrongly, this shows itself when the views are in nested folders. The problem is that the view routes are defined correctly in the router, but the menus are pointing to incorrect path.

**Comment by taefi** [comment]
Fixed with #2462
